README.md

```diff
@@ -74,6 +74,7 @@ src/                         project source code
 |- app/                      app components
 |  |- core/                  core module (singleton services and single-use components)
 |  |- shared/                shared module  (common components, directives and pipes)
+|  |- i18n/                  international language service and component
```

Comment:
I think the internal third-party modules used by application is normal.

The services and components are easier to reuse if they are organized according to business.

The services which putting into the `core` are the "sigleton" pattern.  But it is not only way to do so.

Reply:
I don't think i18n can be qualified as a feature module, as it's a transverse utility without any business value and thus belongs to core/shared module in regard to Angular style guide.

Most projects will adapt the base template anyways, and I would prefer keeping it as close as possible to the Angular style guide.

Reply:
First the translation is the i18n's business.

The i18n service and components together could be easy added or remove to/from an app.
Or you should search the translation feature on the core module and others on the share module.
And even It could be extract as separate library(the only problem is howto be Independent the UI).  But It can be reused on different ionic/md/bootstrap application(Not just as a template application, but also as a library).

The `core` and `share` module should put the services and components only avaiable on the app.
They can not be ported to another app.


eg, I have a Main App and many Sub-App work/compile together(they used/shared the same UI).
The each app could have to a piece of i18n code If the i18n is not a library.

Reply:
I understand the rationale behind it, but I still think it would be better to keep it under existing core/shared module for the *default* generated template.
We try to stand behind the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) whenever we can, and IMHO adding a i18n module here goes too far for a base sample code aiming to be used in any app.

I agree your separation may be useful in some cases as you stated, but the opposite can also be said: in my company apps, the i18n services/components are heavily customized for each app and therefore it does not make sense to separate them from other transverse services.

And if you plan to share some services/components between apps, you should better make it a separate library, you can even use [Nx extensions](https://nrwl.io/nx) to make it in a same repo for your app/sub-apps.

If you want to make this separation the default for you own apps you can still do this by creating an add-on generator, but I don't think it's a suitable default for everyone especially as the Angular Style Guide specifies that such transverse services/components should be put in core/shared.

Reply:
I think this is a demo to teach guys : how to reuse and organize code in an application?

* Application Layer
  * Core/Shared
* Internal Library Layer
* External Library Layer 

BTW The NX is too heavey for me to KISS.